linalg_ops/matmul/CMakeLists.txt

```diff
@@ -1,2036 +1,193 @@
+###############################################################################
+
+set(_DTYPES)
+list(APPEND _DTYPES "i8_into_i32")
+list(APPEND _DTYPES "f32_into_f32")
+list(APPEND _DTYPES "f16_into_f16")
+list(APPEND _DTYPES "f16_into_f32")
+list(APPEND _DTYPES "bf16_into_bf16")
+list(APPEND _DTYPES "bf16_into_f32")
+# list(APPEND _DTYPES "f8E4M3FNUZ_into_f32")  # Unsupported data type.
+foreach(_DTYPE IN LISTS _DTYPES)
+  foreach(_SIZE IN LISTS _SIZES)
+    iree_test_suites_runner_test(
+      NAME
+        matmul_cpu_${_DTYPE}_${_SIZE}
+      TESTS_SRC
+        "generated/${_DTYPE}/matmul_${_DTYPE}_${_SIZE}.mlir"
+      CALLS_SRC
+        "generated/${_DTYPE}/matmul_${_DTYPE}_${_SIZE}_calls.mlir"
+      TEST_RUNNER
+        iree-test-suites_iree-e2e-matmul-test
+      TARGET_BACKEND
+        "llvm-cpu"
+      DRIVER
+        "local-task"
+      COMPILER_ARGS
+      RUNNER_ARGS
+      LABELS
+    )
+  endforeach()
+endforeach()
+
+###############################################################################
+#
+# CPU - vmvx on local-task, default flags.
+#
+###############################################################################
+
+set(_DTYPES)
+list(APPEND _DTYPES "i8_into_i32")
+list(APPEND _DTYPES "f32_into_f32")
+# list(APPEND _DTYPES "f16_into_f16")  # Unsupported data type.
+# list(APPEND _DTYPES "f16_into_f32")  # Unsupported data type.
+# list(APPEND _DTYPES "bf16_into_bf16")  # Unsupported data type.
+# list(APPEND _DTYPES "bf16_into_f32")  # Unsupported data type.
+# list(APPEND _DTYPES "f8E4M3FNUZ_into_f32")  # Unsupported data type.
+foreach(_DTYPE IN LISTS _DTYPES)
+  # Note: not running large tests on vmvx, too slow.
+  set(_SIZE "small")
+  iree_test_suites_runner_test(
+    NAME
+      matmul_vmvx_${_DTYPE}_${_SIZE}
+    TESTS_SRC
+      "generated/${_DTYPE}/matmul_${_DTYPE}_${_SIZE}.mlir"
+    CALLS_SRC
+      "generated/${_DTYPE}/matmul_${_DTYPE}_${_SIZE}_calls.mlir"
+    TEST_RUNNER
+      iree-test-suites_iree-e2e-matmul-test
+    TARGET_BACKEND
+      "vmvx"
+    DRIVER
+      "local-task"
+    COMPILER_ARGS
+    RUNNER_ARGS
+    LABELS
+  )
+endforeach()
+
+###############################################################################
+#
+# GPU - Vulkan, default flags.
+#
+###############################################################################
+
+set(_DTYPES)
+# list(APPEND _DTYPES "i8_into_i32")  # Currently failing.
+list(APPEND _DTYPES "f32_into_f32")
+# list(APPEND _DTYPES "f16_into_f16")  # Failing to compile.
+# list(APPEND _DTYPES "f16_into_f32")  # Failing to compile.
+# list(APPEND _DTYPES "bf16_into_bf16")  # Failing to compile.
+# list(APPEND _DTYPES "bf16_into_f32")  # Failing to compile.
+# list(APPEND _DTYPES "f8E4M3FNUZ_into_f32")  # Unsupported data type.
+foreach(_DTYPE IN LISTS _DTYPES)
+  foreach(_SIZE IN LISTS _SIZES)
+    iree_test_suites_runner_test(
+      NAME
+        matmul_vulkan_${_DTYPE}_${_SIZE}
+      TESTS_SRC
+        "generated/${_DTYPE}/matmul_${_DTYPE}_${_SIZE}.mlir"
```

Comment:
Could pass list of xpass, xfail, skip here... maybe. Would then want to move compilation from build time to test time.

Vulkan should support f16 and i8, but I think that is target/extension dependent (see the `--iree-vulkan-target=valhall` flag in the iree-org/iree Vulkan tests and the `vulkan_uses_vk_khr_shader_float16_int8` label )